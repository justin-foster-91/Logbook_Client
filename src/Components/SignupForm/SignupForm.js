import React, { useState } from 'react';
import AuthApiService from '../../Services/auth-api-service';
import './SignupForm.css';

const SignupForm = (props) => {
  const [error, setError] = useState(null);

  // TODO: I want to move the error message to the SignupPage. Maybe a pop-up tooltip?
  const handleSubmit = (e) => {
    e.preventDefault()
    const { username, password, passwordVerify } = e.target;

    if(password.value !== passwordVerify.value){
      setError("Password do not match");
      return;
    }

    AuthApiService.postUser({
      username: username.value,
      password: password.value,
    })
      .then(user => {
        console.log("New user submitted");
        username.value = ''
        password.value = ''
        passwordVerify.value = ''
        props.onSignupSuccess()
      })
      .catch(res => {
        setError(res.error);
      })
    }
  

    return (
    <>
      <div>
        <form
          className="SignupForm"
          onSubmit={(e) => handleSubmit(e)}
        >
          <label htmlFor="userName">Username: </label><br/>
          <input className='username' type='text' required id='username'></input>

          <label htmlFor="password">Password:</label><br/>
          <input className='password' type='text' required id='password'></input>

          <label htmlFor="passwordVerify">Confirm Password:</label><br/>
          <input className='passwordVerify' type='text' required id='passwordVerify'></input>

          <input type="submit" value="Sign Up" className="formButton"/>
        </form>

      </div>
      <div role='alert'>
        {error ? <p className='red'>{error}</p> : null}
      </div>
    </>
  )
}

SignupForm.defaultProps = {
  onSignupSuccess: () => {}
}

export default SignupForm;