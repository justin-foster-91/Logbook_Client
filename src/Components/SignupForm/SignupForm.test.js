import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SignupForm from './SignupForm';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter><SignupForm /></BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
})

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<BrowserRouter><SignupForm /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});