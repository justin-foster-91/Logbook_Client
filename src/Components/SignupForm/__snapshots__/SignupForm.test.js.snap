// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <div>
    <form
      className="SignupForm"
      onSubmit={[Function]}
    >
      <label
        htmlFor="userName"
      >
        Username: 
      </label>
      <br />
      <input
        className="username"
        id="username"
        required={true}
        type="text"
      />
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <br />
      <input
        className="password"
        id="password"
        required={true}
        type="password"
      />
      <label
        htmlFor="passwordVerify"
      >
        Confirm Password:
      </label>
      <br />
      <input
        className="passwordVerify"
        id="passwordVerify"
        required={true}
        type="password"
      />
      <input
        className="formButton"
        type="submit"
        value="Sign Up"
      />
    </form>
  </div>,
  <div
    role="alert"
  />,
]
`;
